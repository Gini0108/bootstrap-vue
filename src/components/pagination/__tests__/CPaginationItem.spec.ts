import { mount } from '@vue/test-utils'
import { CPaginationItem as Component } from '../../../index'

const ComponentName = 'CPaginationItem'

const defaultWrapper = mount(Component, {
  propsData: {
      
  },
  slots: {
    default: 'Default slot'
  },
})

const customWrapper = mount(Component, {
  propsData: {
    active: true,
    component: 'div',
    disabled: true,
    href: '/bazinga'
  },
  slots: {
    default: 'Default slot'
  },
})

describe(`Loads and display ${ComponentName} component`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.html()).toMatchSnapshot()
  }) 
  it('contain slots and classes', () => {
    expect(defaultWrapper.find('.page-link').text()).toContain('Default slot')
    expect(defaultWrapper.classes('page-item')).toBe(true)
    expect(defaultWrapper.find('.page-link').attributes('href')).toContain('/bazinga')  // TODO: It must be working!
  })
})

describe(`Customize ${ComponentName} component`, () => {
  it('renders correctly', () => {
    expect(customWrapper.html()).toMatchSnapshot()
  }) 
  it('contain slots and classes', () => {
    expect(customWrapper.find('.page-link').text()).toContain('Default slot')
    expect(customWrapper.classes('page-item')).toBe(true)
    expect(customWrapper.classes('active')).toBe(true)
    expect(customWrapper.classes('disabled')).toBe(true)
    expect(customWrapper.attributes('active')).toBe('true')
    expect(customWrapper.attributes('aria-current')).toBe('page')
  })
})