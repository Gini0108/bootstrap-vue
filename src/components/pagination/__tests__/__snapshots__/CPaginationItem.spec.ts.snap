// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customize CPaginationItem component renders correctly 1`] = `
"<li class=\\"page-item active disabled\\" active=\\"true\\" aria-current=\\"page\\">
  <div class=\\"page-link\\">Default slot</div>
</li>"
`;

exports[`Loads and display CPaginationItem component renders correctly 1`] = `"<li class=\\"page-item\\"><a class=\\"page-link\\">Default slot</a></li>"`;
